### PR TITLE
Support tags that have terminating comment

### PR DESCRIPTION
**Pull request by somewhatabstract**
**Summary**

This updates things so that we can support tags that have a start and end comment style. This is especially useful in JSX so that we can write tags using `{\* *\}` type commentary.

### CODE REVIEW

File: src/marker-parser.js [hunk 1]
```diff
@@ -176,30 +185,41 @@ export default class MarkerParser {
          * Groups:
          *     1: The tag id
          */
-        this._endTagDecodeRegExp = new RegExp(`^(\\S+)\\s*$`);
+        this._endTagDecodeRegExp = new RegExp(`^(\\S+)\\s*(\\S*)?$`);
```
Comment (csilvers) [inline src/marker-parser.js:188]: (The `?` doesn't do anything here; you could remove it if you wanted.)

Comment (somewhatabstract) [reply src/marker-parser.js:188]: Oh, yeah, good point.

**Comment by somewhatabstract** [comment]
> Lots of plumbing it looks like.

Exactly. Most of this was just making sure we retained the information. The actual fix part was the regex to extract the info we need to retain and the part where we write it back out when autofixing.

File: src/__tests__/parse-file_test.js [hunk 2]
```diff
@@ -26,22 +28,8 @@ const invokeEvent = (mocked: $Call<typeof jest.fn>, event: string, ...args) => {
...
-        jest.spyOn(fs, "lstatSync").mockReturnValue({isFile: () => true});
-        const pathSpy = jest
-            .spyOn(path, "join")
-            .mockReturnValue("resolved.path");
-        jest.spyOn(path, "normalize").mockReturnValue("normalized.path");
-        const ancesdirSpy = jest
-            .spyOn(Ancesdir, "default")
-            .mockReturnValue("file.dirname");
- */
-
```
Comment (somewhatabstract) [inline src/__tests__/parse-file_test.js:42]: Found and removed some commented code

**Pull request merged**